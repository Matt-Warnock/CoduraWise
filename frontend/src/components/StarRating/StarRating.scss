.star-container {
  --star-size: 60px;
  --star-color: #fff;
  --star-background: #fc0;
  --percent: calc(var(--rating) / 5 * 100%);

  position: relative;
  box-sizing: border-box;
  display: inline-block;
  font-size: 1em;
  font-family: Times; // make sure ★ appears correctly
  line-height: 1;

  &::before {
    content: "★★★★★";
    letter-spacing: 3px;
    background: linear-gradient(
      90deg,
      var(--star-background) var(--percent),
      var(--star-color) var(--percent)
    );
    -webkit-background-clip: text;
    -webkit-text-fill-color: transparent;
  }
}
