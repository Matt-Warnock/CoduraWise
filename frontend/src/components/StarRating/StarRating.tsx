import React from "react";
import "./StarRating.scss";

interface StarRatingProps {
  rating: number;
}

const StarRating = ({ rating }: StarRatingProps) => {
  const style = { "--rating": rating } as React.CSSProperties;

  return (
    <div
      className="star-container"
      style={style}
      aria-label= {`Rating of this product is ${rating} out of 5.`}
    />
  );
};

export default StarRating;
