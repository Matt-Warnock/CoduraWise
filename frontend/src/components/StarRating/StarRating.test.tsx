import { render, screen } from "@testing-library/react";
import StarRating from "./StarRating";
import TestRenderer from "react-test-renderer";
import React from "react";

describe("given the resource has a rating", () => {
  describe("when the rating is one", () => {
    test("then the aria label should convey that", () => {
      const rating = 1;

      render(<StarRating rating={rating} />);

      const stars = screen.getByLabelText(
        `Rating of this product is ${rating} out of 5.`,
      );

      expect(stars).toBeInTheDocument();
    });

    test("snapshot test: the html matches", () => {
      const rating = 1;
      const card = TestRenderer.create(<StarRating rating={rating} />);

      expect(card.toJSON()).toMatchSnapshot();
    });
  });
});

