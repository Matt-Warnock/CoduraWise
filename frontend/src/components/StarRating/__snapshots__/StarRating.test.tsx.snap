// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`given the resource has a rating when the rating is one snapshot test: the html matches 1`] = `
<div
  aria-label="Rating of this product is 1 out of 5."
  className="star-container"
  style={
    Object {
      "--rating": 1,
    }
  }
/>
`;
